drivers/accel/adxl355/adxl355.c

```diff
@@ -924,8 +962,16 @@ static int64_t adxl355_accel_conv(struct adxl355_dev *dev,
  *
  * @return ret     - Converted data.
 *******************************************************************************/
-static int64_t adxl355_temp_conv(uint16_t raw_temp)
+static int64_t adxl355_temp_conv(struct adxl355_dev *dev, uint16_t raw_temp)
 {
-	return ((raw_temp*ADXL355_TEMP_OFFSET_DIV +  ADXL355_TEMP_OFFSET) *
-		(int64_t)ADXL355_TEMP_SCALE_FACTOR);
+	switch(dev->dev_type) {
+	case ID_ADXL355:
+		return ((raw_temp*ADXL355_TEMP_OFFSET_DIV +  ADXL355_TEMP_OFFSET) *
+			(int64_t)ADXL355_TEMP_SCALE_FACTOR);
+	case ID_ADXL359:
+		return ((raw_temp*ADXL359_TEMP_OFFSET_DIV +  ADXL359_TEMP_OFFSET) *
+			(int64_t)ADXL359_TEMP_SCALE_FACTOR);
+	default:
+		return 0;
```

Comment:
`-EINVAL`?

Reply:
in this case the returned value can also be negative. the default case will never be reached. 0 return value in case of an error is suitable in this case.